src/main/java/com/atlassian/db/replica/spi/TransactionHook.java

```diff
@@ -0,0 +1,23 @@
+package com.atlassian.db.replica.spi;
+
+import java.sql.Connection;
+
+public interface TransactionHook {
+    /**
+     * Informs that {@code main} received an UPDATE, INSERT or DELETE or transaction commit
```

Comment:
Doesn't it also inform about `SELECT writingFunction()` and other JDBC non-execute API calls?